rpc_util.go

```diff
@@ -867,21 +870,43 @@ func recvAndDecompress(p *parser, s recvCompressor, dc Decompressor, maxReceiveM
 }
 
 // Using compressor, decompress d, returning data and size.
-// Optionally, if data will be over maxReceiveMessageSize, just return the size.
+// If the decompressed data exceeds maxReceiveMessageSize, it returns nil, 0, and an error.
```

Comment:
We can just mention that the function returns `errMaxMessageSizeExceeded` here, no need to specify other values. See the [styleguide](https://google.github.io/styleguide/go/decisions#returning-errors) section about returning errors:
> If a function returns an error, callers must treat all non-error return values as unspecified unless explicitly documented otherwise

Updated message can be similar to the following:
>  If the decompressed data exceeds maxReceiveMessageSize, it returns errMaxMessageSizeExceeded.